packages/commonwealth/server/federation/index.ts

```diff
@@ -12,6 +13,13 @@ log.info(
       .join(', '),
 );
 
+export const applyCanvasSignedDataMiddleware: (
```

Comment:
This is the one part that can't actually be moved to libs because it needs to have access to a `canvas` instance... I have a note to refactor it into a "middleware factory" that takes a `canvas: CanvasApp` object and returns applyCanvasSignedData/applyCanvasSignedDataMiddleware as a combined function but that would be a pretty big change

Reply:
I see. we can work on a canvas port/adapter solution later I guess.